### PR TITLE
[DOC] Reformatting Ember API docs for Typescript - `@ember/object/observable`

### PR DESCRIPTION
**Pull request by 52052100**
In response to https://github.com/emberjs/ember.js/issues/20172.

Move comments and jsdocs from Mixin.create({...}) to interface for observable.ts.

### CODE REVIEW

**Review by ttbach — APPROVED** [review]

**Review by chriskrycho — CHANGES_REQUESTED** [review]
Thanks so much for doing this! Note that we don't want to just *duplicate* the entries; we want to *move* them. So the existing doc comments on the body of `Mixin.create()` on lines 340 and following need to be removed as well.

**Comment by ttbach** [comment]
> Thanks so much for doing this! Note that we don't want to just _duplicate_ the entries; we want to _move_ them. So the existing doc comments on the body of `Mixin.create()` on lines 340 and following need to be removed as well.

@chriskrycho if we remove existing doc comments, doesn't that remove the comments in prod right now? I wasn't sure, which is why I told @52052100 to do that. Feel free to correct me if I'm wrong.

**Comment by chriskrycho** [comment]
> if we remove existing doc comments, doesn't that remove the comments in prod right now?

Nope, for two reasons:

1. The prod stuff is built and served statically; it’s not re-crawling this every time (that would be :scream:).
2. The comments still exist; they’re just moving to a new location in the file, which will generate the same output when run through YUIDoc.

If we have both, I don’t even know what YUIDoc will do—it’s quite possible and even likely it will end up with *duplicates*, which we definitely don’t want.

**Comment by 52052100** [comment]
> > if we remove existing doc comments, doesn't that remove the comments in prod right now?
> 
> Nope, for two reasons:
> 
> 1. The prod stuff is built and served statically; it’s not re-crawling this every time (that would be 😱).
> 2. The comments still exist; they’re just moving to a new location in the file, which will generate the same output when run through YUIDoc.
> 
> If we have both, I don’t even know what YUIDoc will do—it’s quite possible and even likely it will end up with _duplicates_, which we definitely don’t want.

Thank you for your explanation, will remove the comments inside Mixin.create()

**Review by ttbach — APPROVED** [review]

File: packages/@ember/object/observable.ts [hunk 1]
```diff
@@ -435,15 +280,137 @@ const Observable = Mixin.create({
...
+    @method endPropertyChanges
+    @return {Observable}
+    @private
+  */
+  endPropertyChanges() {
+    endPropertyChanges();
+    return this;
+  },
+
+  /**
```
Comment (chriskrycho) [inline packages/@ember/object/observable.ts:396]: This comment needs to be moved to the interface.

**Review by chriskrycho — CHANGES_REQUESTED** [review]
One missed – let me know when you’ve moved that one and we’ll be ready to merge it. Thanks!

File: packages/@ember/object/observable.ts (hunk 1 above)
Comment (52052100) [reply packages/@ember/object/observable.ts]: I've moved the missing comment, thanks!

**Review by chriskrycho — APPROVED** [review]

**Comment by chriskrycho** [comment]
Thank you! I’ve triggered CI and set it to merge once it's ✅!

**Review by chriskrycho — CHANGES_REQUESTED** [review]
Ah, this won't pass because linting is correctly identifying a gap in the original instructions. You'll need to copy over all the annotations (`@public`, `@method`, etc.)—those are critical parts of how the doc generator tool works. Please incorporate those and then I'll re-review and h*ten* it should be ready to merge.

**Review by chriskrycho — APPROVED** [review]
Thanks for making the update!

**Pull request merged**